### PR TITLE
Cleanup the strategy and sync batch norm.

### PR DESCRIPTION
**Pull request by mingxingtan**
A follow up to https://github.com/google/automl/pull/435

### CODE REVIEW

File: efficientdet/keras/utils_keras_test.py [hunk 1]
```diff
@@ -13,27 +13,28 @@
...
-  )
-  def test_batch_normalization(self, is_training_bn):
+  @parameterized.named_parameters(
+      ('train_local', True, ''), ('eval_local', False, ''),
+      ('train_tpu', True, 'tpu'), ('eval_tpu', False, 'tpu'))
+  def test_batch_norm(self, is_training, strategy):
     inputs = tf.random.uniform([8, 40, 40, 3])
-    with self.subTest(is_training_bn=is_training_bn):
-      expect_results = utils.batch_norm_act(inputs, is_training_bn, None)
-      tf.keras.backend.set_learning_phase(int(is_training_bn))
```
Comment (mingxingtan) [inline efficientdet/keras/utils_keras_test.py:33]: @fsx950223 I believe set_learning_phase is going to be deprecated, so I changed it to directly call bn with is_training.

File: efficientdet/keras/utils_keras.py [hunk 2]
```diff
@@ -30,15 +49,9 @@ def batch_normalization(is_training_bn: bool,
   else:
     gamma_initializer = tf.ones_initializer()
 
-  if data_format == 'channels_first':
-    axis = 1
-  else:
-    axis = -1
-  if strategy in ['tpu', 'gpu'] and is_training_bn:
-    batch_norm_class = tf.keras.layers.experimental.SyncBatchNormalization
```
Comment (mingxingtan) [inline efficientdet/keras/utils_keras.py:38]: I haven't verified tf.keras.layers.experimental.SyncBatchNormalization can work well for TPU or horovod, so maybe we can just use the existing bn for now?

File: efficientdet/keras/utils_keras.py [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
 
-def batch_normalization(is_training_bn: bool,
-                        init_zero: bool = False,
-                        data_format: Text = 'channels_last',
-                        momentum: float = 0.99,
-                        epsilon: float = 1e-3,
-                        strategy: Text = None,
-                        name: Text = 'tpu_batch_normalization'):
-  """Performs a batch normalization followed by a non-linear activation.
+def build_batch_norm(is_training_bn: bool,
```
Comment (mingxingtan) [inline efficientdet/keras/utils_keras.py:24]: I changed batch_normalization to build_batch_norm, because batch_normalization reads like a function that takes a tensor as input and run the batch norm, but here it actually just build the batch_norm_layer.
